### PR TITLE
Separate log file for consistent verbose logs, verbose log file archiving, verbose log file per cycle

### LINKED ISSUE #255: Should verbose mode output to stdout? (continued)
Right now, when using `-V`, the stdout information remains the same. If you want to see the additional verbose messages  you  have to look in app.log. This is atypical of most CLI-based programs. With most GNU apps, as you increase `-v`, the output to stdout increases.  This would help users copy/paste more easily to slack/github as well vs.  having to upload a log file each time.

Thoughts?

### PR DESCRIPTION
**Pull request by habanoz**
This PR resolves #255  . The following steps were performed:

- Verbose output is controlled by -V,--verbose option. It can either be 'on' or 'off'. The default value is 'on', so that new users can collect verbose information in case they experience problems. Experienced users are recommended to turn verbose logging off by setting '-V off' option.

- Verbose log files are created in the logging directory, near the app log file. Verbose logfile names are named in the form of 'app_verbose_{cycle}_%Y%M%D_%H%m%s.log'. At application launch, since no payment cycle is selected, the verbose logging file name is in the form of 'app_verbose_init_%Y%M%D_%H%m%s.log'.

- Old verbose logging files are moved to the 'verbose_backup' folder which is created inside the logging directory. At most 60 backup files are stored. Backup files are kept in zip files to reduce disk usage.

- If a user experiences a problem and needs support he is required to provide 2 verbose logging files. First is the init verbose file which is probably moved to 'verbose_backup' folder and contains detailed information about the application launch. Second is the latest verbose log file which has detailed information for the current cycle. 

* **Check list**:

[X] I extended the Travis CI test units with the corresponding tests for this new feature (if needed).

[ ] I extended the Sphinx documentation (if needed).

[X] I extended the help section (if needed).

[ ] I changed the README file (if needed).

[ ] I created a new issue if there is further work left to be done (if needed).

**Work effort**: 7.5H

### CODE REVIEW

**Comment by jdsika** [comment]
@habanoz you need a green build for someone else to review first. I suggest to use [WIP] in the title to indicate that it is still in progress and not ready for review. Please update the title if my suggestion does not fit in your opinion.

**Comment by habanoz** [comment]
@jdsika 

Thanks for WIP update. I have changed the title. Travis checks are fixed. Ready to merge after review.

**Comment by utdrmac** [comment]
There's A LOT of changes in here that are completely unnecessary. pycodestyle should be ignoring the "long line" rule. @habanoz do you have src/setup.cfg in your branch? This file tells pycodestyle to ignore 2 specific rules so that we don't have to do all that formatting that is in the PR.

**Comment by habanoz** [comment]
> There's A LOT of changes in here that are completely unnecessary. pycodestyle should be ignoring the "long line" rule. @habanoz do you have src/setup.cfg in your branch? This file tells pycodestyle to ignore 2 specific rules so that we don't have to do all that formatting that is in the PR.

@utdrmac Would you mind pointing completely unnecessary changes?

File: src/launch_common.py
Comment (jdsika) [inline src/launch_common.py:121]: @utdrmac are you talking about the style change here?

File: src/launch_common.py
Comment (jdsika) [inline src/launch_common.py:127]: I think those linebreaks are very useful?

**Comment by utdrmac** [comment]
Here is an example:
https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/317/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R38

I wasn't sure if these were intentional changes or the result of pycodestyle saying there is a problem. pycodestyle should be ignoring long lines and should not be forcing this formatting.

**Comment by habanoz** [comment]
> Here is an example:
> https://github.com/tezos-reward-distributor-organization/tezos-reward-distributor/pull/317/files#diff-2e5ad92c43aa96cc3a9cef6c6aec998b216f1379c43b1f651013d25e55989312R38
> 
> I wasn't sure if these were intentional changes or the result of pycodestyle saying there is a problem. pycodestyle should be ignoring long lines and should not be forcing this formatting.

@utdrmac 

First, there are not "A LOT" of such style related changes.  

Second, I had to make the changes because travis checks were failing. Yes, there is "src/setup.cfg" file, the feature branch is a copy of the master branch.

**Comment by utdrmac** [comment]
> Second, I had to make the changes because travis checks were failing.

That's what I'm saying, the checks should not have been failing. You are changing lines of code that have not been needed to be altered by any of the previous 6 PR's that we've merged, all which passed travis checks. This is why I'm confused.

Are you running the checks locally and that's where they are failing? The setup.cfg has very specific rules for ignoring long lines of text.

**Comment by jdsika** [comment]
I am just dropping by to mention that you guys should be aware of potentially different style of communication as I sense a little tension here. I think in a review you should mention as much as possible and be happy that a reviewer will focus on even the smallest details. A bug is much worse then 2 more hours of formatting. In the end you can dry your tears with the extra money for the effort. But quality is king!

**Comment by utdrmac** [comment]
My apologies. I may have come off a bit harsh. I was just surprised to see 34 file changes, a lot of which looked to be simple formatting changes that should not have been enforced by our code formatting tool. I jumped to the conclusion that @habanoz 's environment was misconfigured in some way since pycodestyle has never mentioned any of those changes in the past.
I certainly won't disapprove of a PR because of formatting. Formatting is actually a huge issue with me. :) I like my code "looking" neat.

**Comment by habanoz** [comment]
I also like my code looking good. This is my profession. 

I also choose my words carefully not to annoy my teammates. I am not a native speaker and sometimes I fail to do so. For such situations, I present my humble apologies. 

Let's go back to the discussion. Some of the formatting changes were enforced by Travis ci failures. Commits mentioning "style change" indicate such changes.  

Much of the style changes are pycharm magic, not my intentional modifications. They may seem to be the result of an effort to you, but they are not.  

The number of affected files is high not because of formatting changes but because of the removal of the verbose flag which was passed to all object constructors which is not necessary anymore. Still, formatting changes are numerable and certainly not "A LOT". 

Pycharm formatting is compliant with the standard.  Also, I assume we all are using pycharm. If you are very annoyed with long lines, we can work to change the behavior. I am using "pycharm community 2020.2". There is a setup.cf file present.

**Comment by utdrmac** [comment]
I'm not using pycharm. I use BBEdit and nano for all my text editing. The only code formatting that has been enforced was the use of pycodestyle in our travis script. I've got another PR open for changing pycodestyle to use flake8 which is a combination of pycodestyle + code linting. I did see some tools out there for auto-formatting, but I'm not a fan of those tools that change code without my approval. :)

**Comment by jdsika** [comment]
@amzid fun conflicts due to style check

**Comment by jdsika** [comment]
> 
> 
> I'm not using pycharm. I use BBEdit and nano for all my text editing. The only code formatting that has been enforced was the use of pycodestyle in our travis script. I've got another PR open for changing pycodestyle to use flake8 which is a combination of pycodestyle + code linting. I did see some tools out there for auto-formatting, but I'm not a fan of those tools that change code without my approval. :)

Should we add a section in the documentation on "how to get started coding. Setup the environment, what tools we recommend. Configure build checks?

**Review by jdsika — APPROVED** [review]

**Pull request merged**
